core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -2034,7 +2034,8 @@ private[spark] class DAGScheduler(
     // if the cluster manager explicitly tells us that the entire worker was lost, then
     // we know to unregister shuffle output.  (Note that "worker" specifically refers to the process
     // from a Standalone cluster, where the shuffle service lives in the Worker.)
-    val fileLost = workerHost.isDefined || !env.blockManager.externalShuffleServiceEnabled
+    val fileLost = (workerHost.isDefined || !env.blockManager.externalShuffleServiceEnabled) &&
+      env.blockManager.markFileLostOnExecutorLost
```

Comment:
shouldn't we check this config to not take effect, if `spark.shuffle.manager` is `sort` (the default one)?